### PR TITLE
[CHE-8] Add Klarna

### PR DESCRIPTION
**Pull request by ceriksson**
CHE-

# What this PR does

Adds Klarna (direct + vaulted) payment method to SDK.

# Notable decisions & other stuff

Main API calls are defined in `KlarnaService.swift`. Authentication for this payment method uses a web view (see `OAuthViewModel.swift` and `OAuthViewController.swift`)

# Before merging
Please review the files mentioned above and suggest any changes where relevant.

_QA_

- [x] I manually tested this with a demo application (simulator)
- [x] I manually tested this with a demo application (real device)
- [x] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: PrimerSDK/Classes/Services/Parser/JSON/JSONParser.swift [hunk 1]
```diff
@@ -10,7 +10,10 @@ import Foundation
 class JSONParser: Parser {
     private let jsonDecoder = JSONDecoder()
     
-    func parse<T: Decodable>(_ type: T.Type, from data: Data) throws -> T {
+    func parse<T: Decodable>(_ type: T.Type, from data: Data, keyDecodingStrategy: JSONDecoder.KeyDecodingStrategy? = nil) throws -> T {
+        if let keyDecodingStrategy = keyDecodingStrategy  {
+            jsonDecoder.keyDecodingStrategy = keyDecodingStrategy
```
Comment (EvansPie) [inline PrimerSDK/Classes/Services/Parser/JSON/JSONParser.swift:15]: Let's try and remove this, and refactor where needed

File: PrimerSDK/Classes/User Interface/OAuth/OAuthViewModel.swift [hunk 2]
```diff
@@ -88,34 +88,76 @@ class OAuthViewModel: OAuthViewModelProtocol {
                 switch result {
                 case .failure(let err): completion(err)
                 case .success(let res):
-                    instrument = PaymentInstrument(
```
Comment (EvansPie) [inline PrimerSDK/Classes/User Interface/OAuth/OAuthViewModel.swift:91]: This function is huge with a lot of nested if-else statements. Let's split that up.

**Review by EvansPie — APPROVED** [review]
I'm approving this PR but I'm also pushing some cleanups for you to have a look. No logic has changed.

File: PrimerSDK/Classes/User Interface/OAuth/OAuthViewModel.swift (hunk 2 above)
Comment (ceriksson) [reply PrimerSDK/Classes/User Interface/OAuth/OAuthViewModel.swift:91]: busted! I'll add something a little more sophisticated 😇

**Pull request merged**